diff -urN busybox-1.18.5/include/libbb.h busybox-1.18.5.new/include/libbb.h
--- busybox-1.18.5/include/libbb.h	2011-06-12 23:41:51.000000000 +0800
+++ busybox-1.18.5.new/include/libbb.h	2013-03-19 13:55:26.000000000 +0800
@@ -29,6 +29,7 @@
 #include <sys/poll.h>
 #include <sys/ioctl.h>
 #include <sys/mman.h>
+#include <sys/resource.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
 #include <sys/time.h>
